src/components/action-popover/action-popover.component.js

```diff
@@ -25,6 +28,25 @@ const ActionPopover = ({
   const [guid] = useState(createGuid());
   const button = useRef();
   const menu = useRef();
+  // const elementDOM = useRef(document.createElement("div"));
+  const popperInstance = useRef();
+  const isMounted = useRef(false);
+
+  const valueMapper = useCallback(() => {
```

Comment:
I think this should be `useMemo` as it accepts no arguments. It's debatable if the overhead of `useMemo` or `useCallback` provides any performance improvement when the logic is simple.
```suggestion
  const mappedPlacement = useMemo(() => {
```